mongo/client_test.go

```diff
@@ -344,4 +344,49 @@ func TestClient(t *testing.T) {
 			})
 		}
 	})
+	t.Run("serverAPI version", func(t *testing.T) {
+		t.Run("success", func(t *testing.T) {
+			ServerAPIOptions := options.ServerAPI().SetServerAPIVersion(options.ServerAPIVersion1)
+			client, err := NewClient(&options.ClientOptions{ServerAPIOptions: ServerAPIOptions})
```

Comment:
nit: Instead of manually doing `&options.ClientOptions{...}`, use the builders API:

```
NewClient(options.Client().SetServerAPIOptions(...))
```

These should be the same functionally, but the builders are considered more idiomatic.

Reply:
Nice, I added `clientoptions#SetServerAPIOptions()` as it didn't exist before.